doc/source/user/build_system/dependencies.rst

```diff
@@ -0,0 +1,175 @@
+
+.. code-block:: yaml
+
+   # Excerpt of dualblinky.core
+
+   # ...
+
+   filesets:
+     # ...
+     rtl:
+       # ...
+       depend:
+         - ">=fusesoc:examples:blinky:1.0"
+
+.. note::
+
+   YAML requires quotation marks for strings with special characters, as they are used in version constraints.
+   Both single (``'``) and double quotes ``"`` can be used.
+
+
+File ordering
+-------------
+
+File ordering (compilation order) is important in many hardware design projects.
+The following rules apply.
+
+* Files from dependencies are inserted into the file list before the files in the file set where the dependency is declared.
+* The order in which dependencies are listed in the ``depend`` section does not imply any ordering.
+  That is, specifiying ``depend: [A, B]`` does not guarantee that files from core ``A`` are included before the ones from core ``B``.
+  (If such an order is desired, make core ``B`` depend on ``A``.)
+
+What happens if a dependency is specified?
+------------------------------------------
+
+Declaring a dependency includes the dependent core in the build.
+More specificially, the following sections specified in the ``default`` target of the dependent core are included:
+
+* ``filesets``: File sets to include.
+* ``hooks``: A list of hooks to execute.
+* ``generate``: List of generators.
+* ``parameters``: List of available parameters.
+* ``vpi``: List of VPI objects.
+
+Notably *not* included are the ``tools``, ``toplevel``, ``description``, and ``default_tool`` sections of the ``default`` target.
+Also, no target other than ``default`` is considered when including a dependency.
+
+Version constraints
+-------------------
+
+Version constraints specify which version of a dependent core can be used, and which versions are incompatible.
+
+Within a :term:`core file`, version constraints are expressed by prefixing a core name with a version comparision operator.
+The following version comparison operators are available.
+
+.. list-table:: Version comparison operators
+   :widths: 10 40 50
+   :header-rows: 1
+
+   * - Operator
+     - Meaning
+     - Example
+   * - ``=``
+     - exactly
+     - ``=fusesoc:examples:blinky:1.2``: exactly version 1.2
+   * - ``<``
+     - less (lower) than
+     - ``<fusesoc:examples:blinky:1.2``: any version before 1.2, e.g. 0.9
+   * - ``<=``
+     - at most (less than or equal to)
+     - ``<=fusesoc:examples:blinky:1.2``: at most version 1.2, e.g. 0.9, or 1.2
+   * - ``>=``
+     - at least (greater than or equal to)
+     - ``>=fusesoc:examples:blinky:1.2``: version 1.2, or any newer version, e.g. 1.2, or 10.7
+   * - ``>``
+     - more (higher) than
+     - ``>fusesoc:examples:blinky:1.2``: any version after 1.2, e.g. 1.3, or 10.7
+   * - ``^``
+     - Caret requirement: any version less than the next major version (see below)
+     - ``^fusesoc:examples:blinky:1.2``: >=1.2.0 <1.3.0
+   * - ``~``
+     - Tilde requirement: allow updates to the current version (see below)
```

Comment:
current _minor_ version ??

Reply:
It's more tricky (confusing) than that, the text below describes it in more detail.

Reply:
oh... I see.. Fair enough then.